config.js

```diff
@@ -550,47 +438,174 @@ exports.updateFromArgv = function() {
     }
 
     if (argv.timecontrolsunranked) {
-        for (let e of argv.timecontrolunranked.split(',')) {
+        for (let e of argv.timecontrolsunranked.split(',')) {
             exports.allowed_timecontrols_unranked[e] = true;
         }
     }
 
-    if (argv.greeting) {
-        exports.GREETING = argv.greeting;
-    }
-    if (argv.farewell) {
-        exports.FAREWELL = argv.farewell;
-    }
-    if (argv.rejectnewmsg) {
-        exports.REJECTNEWMSG = argv.rejectnewmsg;
-    }
-
     exports.bot_command = argv._;
 
-    function familyArrayFromGeneralArg(generalArg) {
-        return ["", "unranked", "ranked" ].map(e => generalArg + e);
+    // Show in debug all the exports results
+    for (let k in exports) {
+        if (k === "apikey") {
+            debugLogExports(`    case hidden: ${k} + hidden`);
+        } else {
+            debugLogExports(`    result: ${k} + ${JSON.stringify(exports[k])}`);
+        }
     }
+    debugLogExports("\n");
+
+    // console messages
+    // B - check deprecated features //
+    testDeprecated("komis");
+
+    // C - check Warnings :
+    const rankedUnrankedFamilies = ["boardsizes", "boardsizewidths", "boardsizeheights", "komis",
+        "speeds", "timecontrols", "minhandicap", "maxhandicap", "minmaintimeblitz","minmaintimelive",
+        "minmaintimecorr","maxmaintimeblitz", "maxmaintimelive", "maxmaintimecorr", "minperiodsblitz",
+        "minperiodslive", "minperiodscorr", "maxperiodsblitz", "maxperiodslive", "maxperiodscorr",
+        "minperiodtimeblitz", "minperiodtimelive", "minperiodtimecorr", "maxperiodtimeblitz",
+        "maxperiodtimelive", "maxperiodtimecorr", "minrank", "maxrank", "noautohandicap", "nopause"];
+    checkWarnings(rankedUnrankedFamilies, "nopause");
+    // end of console messages
 
-    function parseMinmaxRankFromNameString(rankArgNameString) {
-        let re = /(\d+)([kdp])/;
-        let results = argv[rankArgNameString].toLowerCase().match(re);
+}
 
-        if (results) {
-            if (results[2] === "k") {
-                exports[rankArgNameString] = 30 - parseInt(results[1]);
-            } else if (results[2] === "d") {
-                exports[rankArgNameString] = 30 - 1 + parseInt(results[1]);
-            } else if (results[2] === "p") {
-                exports[rankArgNameString] = 36 + parseInt(results[1]);
+function familyArrayFromGeneralExportString(generalExportsString) {
+    return ["", "ranked", "unranked"].map(e => generalExportsString + e);
+}
+
+function parseMinmaxRankFromNameString(argvArg, rankArgNameString) {
+    const re = /(\d+)([kdp])/;
+    const results = argvArg.toLowerCase().match(re);
+
+    if (results) {
+        if (results[2] === "k") {
+            argvArg = 30 - parseInt(results[1]);
```

Comment:
I'm confused by this change. Do we no longer need the `exports`? If no, why are we still calling the functions? If yes? Then something broke here.

Reply:
indeed, there's something wrong here :+1: 
in PR #213 , the exports was replaced by a return, so i carelessly replaced it again here by argvArg instead of exports[rankNameString]

fixed, ty :1st_place_medal: